### DISCUSSION TITLE
Exporting a Sum Metric for Prometheus

### DISCUSSION THREAD

**Discussion by michael-cybrid** [open-telemetry/opentelemetry-ruby#1778] [Q&A]
I'd like to export a counter metric, via opentelemetry-collector, to prometheus but am running into issues.

The docs seem to indicate that only delta sums are support but if I define `OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE` as `cumulative`, the exported data points have that set properly.

What I'm running into is that I don't see how I can create a counter with `is_monotonic` set, which looks to be a prerequisite for Prometheus ingestion. I can see in other languages that this is supported. Is there a way to do this?

**Comment by xuan-cao-swi** [comment]
Hi @michael-cybrid , as current counter implementation (and sum aggregation), there is no feature that support monotonic operation. I have opened an [issue](https://github.com/open-telemetry/opentelemetry-ruby/issues/1783) about this feature enhancement.
